pyuvdata/uvdata/tests/test_fhd.py

```diff
@@ -314,74 +334,46 @@ def test_read_fhd_warnings():
-    assert str(cm.value).startswith(
-        "No recognized key for visibility weights in flags_file."
-    )
 
-    # try cases with extra files of each type
-    extra_xx_file = testdir + testfile_prefix + "extra_vis_XX.sav"
-    copyfile(testfiles[1], extra_xx_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_xx_file])
-    assert str(cm.value).startswith("multiple xx datafiles in filelist")
-    os.remove(extra_xx_file)
 
-    extra_yy_file = testdir + testfile_prefix + "extra_vis_YY.sav"
-    copyfile(testfiles[3], extra_yy_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_yy_file])
-    assert str(cm.value).startswith("multiple yy datafiles in filelist")
-    os.remove(extra_yy_file)
-
-    xy_file = testdir + testfile_prefix + "vis_XY.sav"
-    extra_xy_file = testdir + testfile_prefix + "extra_vis_XY.sav"
-    copyfile(testfiles[1], xy_file)
-    copyfile(testfiles[1], extra_xy_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [xy_file, extra_xy_file])
-    assert str(cm.value).startswith("multiple xy datafiles in filelist")
-    os.remove(xy_file)
-    os.remove(extra_xy_file)
-
-    yx_file = testdir + testfile_prefix + "vis_YX.sav"
-    extra_yx_file = testdir + testfile_prefix + "extra_vis_YX.sav"
-    copyfile(testfiles[1], yx_file)
-    copyfile(testfiles[1], extra_yx_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [yx_file, extra_yx_file])
-    assert str(cm.value).startswith("multiple yx datafiles in filelist")
-    os.remove(yx_file)
-    os.remove(extra_yx_file)
-
-    extra_params_file = testdir + testfile_prefix + "extra_params.sav"
-    copyfile(testfiles[2], extra_params_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_params_file])
-    assert str(cm.value).startswith("multiple params files in filelist")
-    os.remove(extra_params_file)
-
-    extra_flags_file = testdir + testfile_prefix + "extra_flags.sav"
-    copyfile(testfiles[0], extra_flags_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_flags_file])
-    assert str(cm.value).startswith("multiple flags files in filelist")
-    os.remove(extra_flags_file)
-
-    extra_layout_file = testdir + testfile_prefix + "extra_layout.sav"
-    copyfile(testfiles[6], extra_layout_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_layout_file])
-    assert str(cm.value).startswith("multiple layout files in filelist")
-    os.remove(extra_layout_file)
-
-    extra_settings_file = testdir + testfile_prefix + "extra_settings.txt"
-    copyfile(testfiles[7], extra_settings_file)
-    with pytest.raises(ValueError) as cm:
-        fhd_uv.read(testfiles + [extra_settings_file])
-    assert str(cm.value).startswith("multiple settings files in filelist")
-    os.remove(extra_settings_file)
+@pytest.mark.parametrize(
```

Comment:
Nice parameterization to make tests more atomic 👍

Reply:
Going for that 1k test mark!